### ISSUE TITLE
CAPG cluster autoscaling

### ISSUE THREAD

**Issue by alex-dabija** [giantswarm/roadmap#1376]
## Story

_*-As a cluster admin, I want a CAPG cluster to autoscale depending on the required resources in order to ensure the stability of applications running on the cluster.*_

## Background

CAPG cluster don't have support for machine pools, which means we can't leverage the cloud provider's support for autoscaling. Machine deployments are supported and `cluster-autoscaler` does have support for them.

Giant Swarm has been running the `cluster-autoscaler` on the workload cluster in order to enable the cluster to still scale in case the communication between the management cluster and the workload cluster is severed.

Unfortunately, the `cluster-autoscaler` needs to run on the management cluster in order to be able to update the workload cluster's machine deployement resources.

## Resources

- [Cluster Autoscaler documentation](https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/cloudprovider/clusterapi/README.md);

**Comment by alex-dabija** [comment]
We don't have any special configuration for the Kubernetes scheduler to optimize for better usage. We do have the vertical pod autoscaler running by default on vintage clusters which might have an impact on how many pods are running on a node.

This story is only meant to get the cluster autoscaler running on the management cluster to handle autoscaling of machine deployments.

I wouldn't worry for now about packing pods more efficiently on the nodes. I would just keep things simple for now.

**Comment by gawertm** [comment]
@alex-dabija I changed ownership from Hydra to Rocket, as most likely Rocket will implement first. We are already looking at the documentation and trying it out

**Comment by gawertm** [comment]
blocked until https://github.com/giantswarm/giantswarm/issues/23820

**Comment by brinker211** [comment]
@alex-dabija I see the new tickets for CAPA and CAPZ. Is there still a ticket for CAPG? This was the provider this ticket originally started with.

**Comment by brinker211** [comment]
@gawertm is the referenced internal ticket https://github.com/giantswarm/giantswarm/issues/23820 for CAPG? I see Hydra and Clippy referenced for CAPA and CAPZ but looking for the status of CAPG as well. Thanks!

**Comment by gawertm** [comment]
hi @brinker211 yes this was for CAPG but also for the Rocket providers. We initially planned to help Hydra with that. Eventhough its not a Rocket priority anymore

**Comment by gawertm** [comment]
@Rotfuks would the autoscaler topic go Turtles? lets move it form rocket backlog then :)
cc @alex-dabija

**Comment by alex-dabija** [comment]
> @Rotfuks would the autoscaler topic go Turtles? lets move it form rocket backlog then :) cc @alex-dabija


Yes, Turtles makes sense. It would be great to have a common autoscaling solution for all providers (at least for CAPZ, CAPV, CAPVCD). CAPA is still using machine pools which require a different implementation.

**Comment by alex-dabija** [comment]
@Rotfuks I would keep this issue scoped to autoscaling for machine deployments because of our current focus on CAPA. I moved the https://github.com/giantswarm/roadmap/issues/2692 to https://github.com/giantswarm/roadmap/issues/1798. It should make it easier to track the status of CAPA for our first [release](https://github.com/giantswarm/roadmap/issues/2726).
